dwh/query_lib.py

```diff
@@ -268,6 +310,33 @@ def find_patient_aggregates(self, base_patient_url: str) -> pandas.DataFrame:
         'max_value', 'min_date', 'max_date', 'first_value', 'last_value',
         'first_value_code', 'last_value_code']]
 
+  def get_patient_encounter_view(self) -> pandas.DataFrame:
+    """See super-class doc."""
+    self._make_sure_spark()
+    self._make_sure_patient()
+    self._make_sure_encounter()
+    if not self._flat_enc:
+      temp_df = self._enc_df
+      flat_df = temp_df.select(
+          'subject', 'id', 'location', 'type', 'period').withColumn(
+          'locationFlat', F.explode_outer('location')).withColumn(
+          'typeFlat', F.explode_outer('type'))
+      self._flat_enc = flat_df.select(
+          'id', F.col('subject.patientId').alias('patientId'),
+          F.col('locationFlat.location.LocationId').alias('locationId'),
+          F.col('typeFlat.coding.system').alias('typeSystem'),
+          F.col('typeFlat.coding.code').alias('typeCode'),
```

Comment:
encounterTypeSystem / encTypySystem and encounterTypeCode / encTypyCode is a better name

Reply:
Done.